components/Ad.vue

```diff
@@ -17,6 +17,8 @@ function gatewayURL(url) {
     url = 'https://gateway.ipfs.io/ipfs/' + url.slice(7);
   } else if (url.startsWith('https://gateway.pinata.cloud/ipfs/')) { // People aren't paying for their pinata bandwidth...
     url = 'https://gateway.ipfs.io/ipfs/' + url.slice(34);
+  } else if (url.startsWith('file:')) { // Disallowed by our CSP, but block it here too.
```

Comment:
I think we want to `url.toLowerCase()` in this function in general (commenting here since the right place is outside this diff